
From: Akinobu Mita <amgta@yacht.ocn.ne.jp>

If the first kernel enabled CONFIG_DISCONTIGMEM, the second kernel could
not boot.  since crash_reserve_bootmem() never called anywhere.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/i386/mm/discontig.c |    4 ++++
 1 files changed, 4 insertions(+)

diff -puN arch/i386/mm/discontig.c~kdump-config_discontigmem-fix arch/i386/mm/discontig.c
--- 25/arch/i386/mm/discontig.c~kdump-config_discontigmem-fix	2005-01-10 21:15:12.417565824 -0800
+++ 25-akpm/arch/i386/mm/discontig.c	2005-01-10 21:15:12.421565216 -0800
@@ -32,6 +32,7 @@
 #include <asm/e820.h>
 #include <asm/setup.h>
 #include <asm/mmzone.h>
+#include <asm/crash_dump.h>
 #include <bios_ebda.h>
 
 struct pglist_data *node_data[MAX_NUMNODES];
@@ -367,6 +368,9 @@ unsigned long __init setup_memory(void)
 		}
 	}
 #endif
+
+	crash_reserve_bootmem();
+
 	return system_max_low_pfn;
 }
 
_
